### PR TITLE
fix: append query params to existing query params if any

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Append query params to existing query params if any

#### Issue #, if available
Maplibre will parse the urls defined in the style-descriptor file and fetch the tiles. However, in the transform function, query params were added to url starting with `x-amz-user-agent` as the first param. This is an issue when the input url already has a query param. With the fix, `x-amz-user-agent` is appended to existing query params.

#### Description of how you validated changes
Tested with raster tiles fetched from style descriptor

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

File: __tests__/AmplifyMapLibreRequest.test.ts [hunk 1]
```diff
@@ -63,6 +63,26 @@ describe("AmplifyMapLibreRequest", () => {
     expect(request.url).toContain("x-amz-user-agent");
   });
 
+  test("transformRequest appends query params to existing query params if any", () => {
```
Comment (thaddmt) [inline __tests__/AmplifyMapLibreRequest.test.ts:66]: can we add an explicit check to this test that we don't have multiple `?` 

Can we also add test cases for edge cases like
- nothing after the question mark
- nothing after an ampersand
- query params with ampersands

Otherwise change looks good though :)

**Review by thaddmt — DISMISSED** [review]

File: __tests__/AmplifyMapLibreRequest.test.ts (hunk 1 above)
Comment (sreeramsama) [reply __tests__/AmplifyMapLibreRequest.test.ts:66]: Added tests to cover these cases:
- [x] nothing after the question mark
- [x] nothing after an ampersand
- [x] query params with ampersands

**Review by thaddmt — APPROVED** [review]
